broker/src/tunneldigger_broker/broker.py

```diff
@@ -77,6 +83,32 @@ def create_tunnel(self, broker, address, uuid, remote_tunnel_id, client_features
         else:
             tunnel_str = "(%s)" % uuid
 
+        # Rate limit creation of new tunnels with the same IP address.
+        # Runs broker.tunnel-connection-threshold hook if threshold exceeded:
+        if self.connection_rate_limit_per_ip_count > 0 and self.connection_rate_limit_per_ip_time > 0:
+            if address[0] not in self.last_tunnel_created_per_ip:
+                # Create deque collection if IP address does not exist in dict
+                self.last_tunnel_created_per_ip[address[0]] = deque([], self.connection_rate_limit_per_ip_count)
+            tunnelCollection = self.last_tunnel_created_per_ip[address[0]]
+            if len(tunnelCollection) >= self.connection_rate_limit_per_ip_count:
+                scale_delta = now - tunnelCollection[0] # Delta of oldest timestamp in collection and now
+                if scale_delta <= self.connection_rate_limit_per_ip_time:
+                    logger.info("Tunnel {0} connection threshold exceeded ({1} in {2} ({3}) seconds). Running hook".format(
```

Comment:
I don't think "5 in 26 (30) seconds" is very comprehensible.
Also, shouldn't it be `connection_rate_limit_per_ip_count+1`? There are `connection_rate_limit_per_ip_count` previously recorded attempts, this is the next one.